lib/semian/resource.rb

```diff
@@ -2,14 +2,20 @@ module Semian
   class Resource #:nodoc:
     attr_reader :tickets, :name
 
-    def initialize(name, tickets:, permissions: 0660, timeout: 0)
+    class << Semian::Resource
+      # Ensure that there can only be one resource of a given type
+      def instance(*args)
+        Semian.resources[args.first] ||= new(*args)
+      end
+    end
```

Comment:
Did you mean in the comments, or here?

I'll do it here for now.

Because we call instance instead of new, we can detect if a worker for a given resource has already been registered, and provide that instead of creating a new semian resource.

If that resource is deleted or never existed,  a new one would be created instead.